mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/internal/gitlab/core/GitlabRepositoryConnector.java

```diff
@@ -119,18 +118,9 @@ private boolean ignoredProperty(String propertyName) {
 		return false;
 	}
 
-	private final PropertyChangeListener repositoryChangeListener4ConfigurationCache = new PropertyChangeListener() {
+	private PropertyChangeListener repositoryChangeListener4ConfigurationCache = null;
```

Comment:
Is this really a result of clean up? It is strange to see it mutable and nullable instead of the recent final non-nullable.

Reply:
Yes, if look strange but  the Clean Up generates an error.

Cannot reference a field before it is defined. in File GitlabRepositoryConnector.java line 315.

this.clientCache.invalidate(new RepositoryKey(taskRepository));
is changed to
clientCache.invalidate(new RepositoryKey(taskRepository));

Reply:
This Clean Up implementation looks weird: making field mutable cannot be claimed as "Clean Up".

Reply:
However, if you can make them `final` back, this one is good to go.

Reply:
> However, if you can make them `final` back, this one is good to go.

No that is not possible we need to have repositoryChangeListener4ConfigurationCache and repositoryChangeListener4ClientCache none final because we otherwise get an  error

`The blank final field ... may not have been initialized.`

We have configured the Save Actions to do a Clean up and we can not make any changes in file  GitlabRepositoryConnector.java because of the error in (remove `this.`)

```
	private final PropertyChangeListener repositoryChangeListener4ClientCache = evt -> {
		TaskRepository taskRepository = (TaskRepository) evt.getSource();
		clientCache.invalidate(new RepositoryKey(taskRepository));
	};
```

Reply:
Why would one explicitly initialize to null when that is implicit?  If you check in a debugger, that actually generates code to assign the null to field that is already null.

Reply:
> Why would one explicitly initialize to null when that is implicit? If you check in a debugger, that actually generates code to assign the null to field that is already null.

Good catch, I just ran the unit tests and could not find this error.
Next commit will fix this.